src/lib.rs

```diff
@@ -454,7 +489,21 @@ impl<'object, Endian> DebugInfo<'object, Endian>
                 // matches us better. We need to keep going until we find the best one.
             }
 
-            return Ok(Some((path, line, func.map(|u| u.0.name.to_string_lossy()))));
+            let func = func.map(|u| {
+                if self.with_demangling {
+                    // First try demangling as a C++ symbol, then fall back to
+                    // Rust symbols.
```

Comment:
Should we be using the compilation unit's `DW_AT_language` attribute to determine which demangling to use instead?